app/views/answers/_edit.html.erb

```diff
@@ -0,0 +1,9 @@
+<%= form_for answer, remote: true, url: {action: :update, controller: :answers},:html => {:class => "answer-edit-form"} do |f| %>
```

Comment:
Rails should be able to infer the correct object from the answer - you shouldn't need the url - if it's saved you'll get the update path as action without specifying it.